app/src/main/java/com/example/bigowlapp/activity/ViewAnotherUserActivity.java

```diff
@@ -1,112 +1,133 @@
 package com.example.bigowlapp.activity;
 
 import android.os.Bundle;
-import android.widget.ArrayAdapter;
+import android.view.View;
 import android.widget.Button;
-import android.widget.Toast;
+import android.widget.TextView;
 
-import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.lifecycle.LiveData;
 
 import com.example.bigowlapp.R;
+import com.example.bigowlapp.model.SupervisionRequest;
 import com.example.bigowlapp.model.User;
-import com.google.android.gms.tasks.OnCompleteListener;
-import com.google.android.gms.tasks.Task;
-import com.google.firebase.auth.FirebaseAuth;
-import com.google.firebase.auth.FirebaseUser;
-import com.google.firebase.database.DatabaseReference;
-import com.google.firebase.database.FirebaseDatabase;
-import com.google.firebase.firestore.FirebaseFirestore;
-import com.google.firebase.firestore.QueryDocumentSnapshot;
-import com.google.firebase.firestore.QuerySnapshot;
-
-import java.util.HashMap;
-import java.util.Objects;
-import com.example.bigowlapp.model.User;
-//import com.google.firebase.auth.*;
+import com.example.bigowlapp.repository.AuthRepository;
+import com.example.bigowlapp.repository.NotificationRepository;
+import com.google.firebase.Timestamp;
+
+import java.util.List;
+
 
 public class ViewAnotherUserActivity extends AppCompatActivity {
-    /*String otherUserID = getIntent().getStringExtra("userID");*/
     String otherUserID;
+    String currentUserID, noteText;
     User otherUser;
-    String supRequestStatus = "none";
-    String supBtn1 = "Send a request to supervise user";
-    String supBtn2 = "You are supervising this user";
-    String supBtn3 = "Cancel this request to supervise";
-    DatabaseReference supRequestRef;
-    FirebaseAuth mAuth;
-    FirebaseUser mUser;
+    boolean aRequestAlready;
+    boolean shouldCancelRequest = false;
+    String supBtnSend = "Send a request to supervise this user";
+    String supBtnAlready = "You are supervising this user";
+    String supBtnCancel = "You have sent a request already. Cancel request";
+    String canNotSend = "Can not send ";
     private Button supRequestBtn;
-    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
-    String otherUserNumber = getIntent().getStringExtra("smsNumber");
-    /*String contactNumber = getIntent().getStringExtra("contactNumber");*/
+    private AuthRepository authRepository = new AuthRepository();
+    String requestUID;
+    private TextView noteTv;
+    private TextView noteTv2;
```

Comment:
I feel like these variables should be renamed. Not sure what they a refering to, and when variables are being named like `noteTv2`, its quite confusing. Comment applies to the related xml as well.

Reply:
changes have been made